tree/ntuple/v7/test/ntuple_basics.cxx

```diff
@@ -808,6 +808,58 @@ TEST(REntry, Basics)
    EXPECT_NE(&pt, e->GetPtr<void>("pt").get());
 }
 
+TEST(REntry, SubFields)
+{
+   FileRaii fileGuard("test_rentry_subfields");
+   {
+      auto model = RNTupleModel::Create();
+      model->MakeField<CustomStruct>("struct", CustomStruct{1.f, {2.f, 3.f}, {{4.f}, {5.f, 6.f, 7.f}}, "foo"});
+
+      auto ntuple = RNTupleWriter::Recreate(std::move(model), "ntuple", fileGuard.GetPath());
+      ntuple->Fill();
+   }
+
+   auto model = RNTupleModel::Create();
+   auto fldStruct = RFieldBase::Create("struct", "CustomStruct").Unwrap();
+   auto fldA = RFieldBase::Create("a", "float").Unwrap();
+
+   model->AddSubField(std::move(fldA), *fldStruct);
+
+   try {
+      auto tmpModel = model->Clone();
+      auto fldInvalidSubField = RFieldBase::Create("doesnotexist", "float").Unwrap();
+      tmpModel->AddSubField(std::move(fldInvalidSubField), *fldStruct);
+      FAIL() << "should not be able to add a subfield not present in the provided parent";
+   } catch (const RException &err) {
+      EXPECT_THAT(err.what(), testing::HasSubstr("subfield \"doesnotexist\" not found"));
+   }
+
+   model->AddField(std::move(fldStruct));
+
+   auto &entry = model->GetDefaultEntry();
```

Comment:
the test should also exercise a non-default entry (doesn't `RNTupleModel::CreateEntry` need adapting?)